subsys/net/l2/openthread/Kconfig

```diff
@@ -189,6 +189,23 @@ endchoice
 config OPENTHREAD_MBEDTLS
 	bool
 	select MBEDTLS
+	imply MBEDTLS_PSA_CRYPTO_C
+	select MBEDTLS_USE_PSA_CRYPTO
+	select PSA_WANT_ALG_ECDH
+	select PSA_WANT_ALG_ECDSA if OPENTHREAD_ECDSA
+	select PSA_WANT_ALG_JPAKE
+	select PSA_WANT_ALG_GCM
+	select PSA_WANT_ALG_CMAC
+	select PSA_WANT_ALG_ECB_NO_PADDING
+	select PSA_WANT_ALG_HMAC
+	select PSA_WANT_ALG_SHA_256
+	select PSA_WANT_ECC_SECP_R1_256
+	select PSA_WANT_KEY_TYPE_HMAC
+	select PSA_WANT_KEY_TYPE_AES
+	select PSA_WANT_KEY_TYPE_ECC_KEY_PAIR_IMPORT
+	select PSA_WANT_KEY_TYPE_ECC_KEY_PAIR_EXPORT
+	select PSA_WANT_KEY_TYPE_ECC_KEY_PAIR_GENERATE
+	select PSA_WANT_KEY_TYPE_ECC_KEY_PAIR_BASIC
```

Comment:
I'm not an expert of OpenThread, but these `select` miss the `if` conditions that are applied to the legacy Mbed TLS crypto. For example here you have
`select PSA_WANT_ALG_JPAKE`
but the legacy counterpart is:
`select MBEDTLS_ECJPAKE_C if OPENTHREAD_COMMISSIONER || OPENTHREAD_JOINER`

I think these `if` conditions are necessary to avoid extra code to be brought into the final binary. Wdyt?